lib/tsconfig-parser.js

```diff
@@ -104,7 +104,18 @@ module.exports = function calculateProjectParserOptions(code, options, extra) {
     watchCompilerHost.onUnRecoverableConfigFileDiagnostic = diagnosticReporter;
 
     // ensure process doesn't emit programs
-    watchCompilerHost.afterProgramCreate = undefined;
+    watchCompilerHost.afterProgramCreate = program => {
+      // report error if there are any errors in the config file
+      const configFileDiagnostics = program
+        .getConfigFileParsingDiagnostics()
+        .filter(
+          diag =>
+            diag.category === ts.DiagnosticCategory.Error && diag.code !== 18003
+        );
+      if (configFileDiagnostics.length > 0) {
+        diagnosticReporter(configFileDiagnostics[0]);
```

Comment:
Shouldn't this be configFileDiagnostics.forEach(diagnosticReporter); ?

Reply:
As far as this package is concerned, the reporter is going to throw an uncaught error, so it didn't seem to really matter.

Reply:
But in that case, I should just use `ts.formatDiagnostics` or similar rather than just reporting the first.

Reply:
I think I'm just going to leave this as-is for now, and if the ESLint reviewers want all the errors instead of just the first, then I can change it.